mirascope/chat/parsers/openai_parser.py

```diff
@@ -0,0 +1,60 @@
+"""Classes for using parsers with Chat APIs."""
+from typing import Callable, Generator, Optional, Type, Union
+
+from openai.types.chat.chat_completion_message_tool_call import (
+    ChatCompletionMessageToolCall,
+)
+from pydantic import BaseModel
+
+from ..tools import OpenAITool
+from ..types import OpenAIChatCompletionChunk
+from .utils import (
+    append_tool_call_arguments,
+    append_tool_call_function_name,
+    created_new_tool_call,
+    find_tool_class,
+)
+
+
+class OpenAIToolStreamParser(BaseModel):
+    """A utility class to parse `OpenAIChatCompletionChunk`s into `OpenAITools`."""
+
+    tool_calls: list[ChatCompletionMessageToolCall] = []
+    tools: list[Union[Callable, Type[OpenAITool]]] = []
+    _current_tool_type: Optional[Type[OpenAITool]] = None
```

Comment:
does this need to be a property now, or can it go back to being a local variable of the `from_stream` function?

Reply:
good catch